### ISSUE TITLE
EN_setcurve fails to update pump curve coefficients

### ISSUE THREAD

**Issue by LRossman** [OpenWaterAnalytics/EPANET#550]
When `EN_setcurve `applies a new set of data points to a curve that represents a power function pump head curve (`H = a + b * Q^c`), the coefficients of the curve (`a, b, c`) that are stored with the pump object that uses the curve never get updated.

I can think of two ways to remedy this:

1. Have `EN_setcurve `check each pump to see if the pump uses the curve and if so, call `EN_setheadcurve `to re-assign the same curve to the pump which will cause the pump's head curve parameters to be updated.
2. Have the calculation of each pump's head curve parameters occur under `EN_inithyd`, just like the code currently does for pipe resistances.

Any preference between these (I'm favoring 2) or are there any better solutions?

**Comment by samhatchett** [comment]
I would slightly favor #1 for uses where we would adjust pump characteristics during a simulation - so as to avoid re-initializing every time.

**Comment by eladsal** [comment]
I think that 2 is more elegant and can't really see the use case of adjusting the pump curve during the simulation. However, @samhatchett , would updating the curve and then calling `EN_setheadcurveindex` work for your case?

**Comment by LRossman** [comment]
I now agree with @samhatchett that option `1` is preferrable. Use cases include really long (multi-year) simulations that model changes in pump behavior over time and "online" simulations that re-calibrate pump curves in real time. Having toolkit clients call `EN_setheadcurveindex `whenever `EN_setcurve `is called is not ideal since it requires them to write their own code to see which pump(s) use the curve in question, and invalid results will be computed if they fail to do so. Better to have the toolkit automatically do it for them. BTW, `EN_setcurvevalue `has the same bug issue as `EN_setcurve `does.

**Comment by eladsal** [comment]
Interesting use cases. 1 it is.

**Comment by jamesuber** [comment]
Yes our use cases are long term (potentially continuous) simulations where you want to do things like change pump characteristics on an event basis. I suppose that another use case might be a design optimization problem where you are trying to select optimal pump characteristics? I would think an API user would assume that changing the curve would implement those changes in the pump behavior...

**Comment by LRossman** [comment]
Option `1` has been implemented in PR #551 .
